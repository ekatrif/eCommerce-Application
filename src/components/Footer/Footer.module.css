footer {
  position: relative;
  color: white;
  background-color: #1F2827;
  padding: 1em;
}

.footer {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.rs_school {
  width: min(10vw, 125px);
}

.rs_school img {
  filter: invert(1);
}

.page_up {
  position: absolute;
  width: min(5vw, 50px);
  height: min(5vw, 50px);
  right: 1vw;
  top: 1vw;
}

.authors {
  font-weight: bold;
}

.authors ul {
  margin-top: 1vh;
}

.authors ul li a {
  color: #b1b1b1;
  text-decoration: none;
}

.authors ul li a::before {
  content: ' ';
  background: url('../../assets/svg/gh.svg');
  background-size: contain;
  display: inline-block;
  width: 1em;
  height: 1em;
  margin-right: 3px;
}

@media (min-width: 768px) {
  .authors ul li a:hover {
    color: #fff;
  }
}