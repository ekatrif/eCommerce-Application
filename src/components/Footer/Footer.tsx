import React from 'react';
import classes from './Footer.module.css';
import Container from '../UI/container/Container';

const Footer: React.FC = () => {
  const authorText = 'created by:';
  return (
    <footer>
      <Container>
        <div className={classes.footer}>
          <div className={classes.authors}>
            {authorText} <br />
            <ul>
              <li>
                <a href="https://github.com/ekatrif">ekatrif</a>
              </li>
              <li>
                <a href="https://github.com/sheritsh">sheritsh</a>
              </li>
              <li>
                <a href="https://github.com/montek1o">montek1o</a>
              </li>
            </ul>
          </div>
          <div className={classes.copyright}>Tourist Tracks Store Â©2023</div>
          <div className={classes.rs_school}>
            <a href="https://rs.school/js/">
              <img src="../../src/assets/svg/rs_school_js.svg" alt="RS_School_logo" width="100%"></img>
            </a>
          </div>
        </div>
        <div className={classes.page_up}>
          <a href="#">
            <img src="../../src/assets/svg/arrow_up.svg" alt="page_up" width="100%"></img>
          </a>
        </div>
      </Container>
    </footer>
  );
};

export default Footer;
