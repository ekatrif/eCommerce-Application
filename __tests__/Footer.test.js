import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../src/components/Footer/Footer';

test('Footer renders correctly', () => {
  const tree = renderer
    .create(<Footer />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
