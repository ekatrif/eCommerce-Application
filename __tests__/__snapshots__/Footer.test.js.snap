// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <div
    className="container"
  >
    <div
      className="footer"
    >
      <div
        className="authors"
      >
        created by:
         
        <br />
        <ul>
          <li>
            <a
              href="https://github.com/ekatrif"
            >
              ekatrif
            </a>
          </li>
          <li>
            <a
              href="https://github.com/sheritsh"
            >
              sheritsh
            </a>
          </li>
          <li>
            <a
              href="https://github.com/montek1o"
            >
              montek1o
            </a>
          </li>
        </ul>
      </div>
      <div
        className="copyright"
      >
        Tourist Tracks Store Â©2023
      </div>
      <div
        className="rs_school"
      >
        <a
          href="https://rs.school/js/"
        >
          <img
            alt="RS_School_logo"
            src="../../src/assets/svg/rs_school_js.svg"
            width="100%"
          />
        </a>
      </div>
    </div>
    <div
      className="page_up"
    >
      <a
        href="#"
      >
        <img
          alt="page_up"
          src="../../src/assets/svg/arrow_up.svg"
          width="100%"
        />
      </a>
    </div>
  </div>
</footer>
`;
